hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java

```diff
@@ -112,27 +109,20 @@
     @Thin
     private List<Set<ConditionEval>> resolvedEvalSets;
 
-    /*
-     * This should be initialized to the owning trigger's context. It is not set automatically so as to allow
-     * for flexibility.  Note, this is not marked as Thin, whereas the trigger is Thin.
-     */
-    @JsonInclude(Include.NON_EMPTY)
-    private Map<String, String> context;
-
     public Alert() {
         // for json assembly
     }
 
-    public Alert(String tenantId, String triggerId, Severity severity, List<Set<ConditionEval>> evalSets) {
+    public Alert(String tenantId, Trigger trigger, Severity severity, List<Set<ConditionEval>> evalSets) {
         this.tenantId = tenantId;
-        this.triggerId = triggerId;
+        this.trigger = trigger;
         this.severity = (null == severity) ? Severity.MEDIUM : severity;
         this.evalSets = evalSets;
 
         this.ctime = System.currentTimeMillis();
         this.status = Status.OPEN;
 
-        this.alertId = tenantId + "-" + triggerId + "-" + ctime;
+        this.alertId = tenantId + "-" + trigger.getId() + "-" + ctime;
```

Comment:
Perhaps some validation that trigger == null is not permitted ? an IllegalArgumentException sounds good.